---
title: "EB002AS1_bulk_EBalmas_SBianchi"
output: html_document
---
#environment
```{r}
#libraries
library(tidyr)
library(stringr)
library(grid)
library(viridis)
library(ggthemes)
library(dplyr)
library(tidyverse)
library(ggsignif)
library(umap)
library(heatmap3)
library(plyr)
library(edgeR)
library(compareGroups)
library(dbscan)
library(reshape2)
library(scran)
library(Matrix)
library(fgsea)
library(msigdbr)
library(RNAseqQC)
library(DESeq2)
library(ensembldb)
library(tibble)
library(purrr)
library(magrittr)
library(ggplotify)
library(limma)
library(Glimma)
library(topGO)
library(org.Hs.eg.db)
library(ggrepel)
library(igraph)
library(readxl)
library(clusterProfiler)
library(enrichplot)
library(SeuratObject)
library(Seurat)
library(pathview)
library(pheatmap)
library(ggplot2)
library(reshape2)
library(WGCNA)

update_geom_defaults("point", aes(size = 4))
hs = org.Hs.eg.db

#functions
cal_z_score = function(x){(x - mean(x)) / sd(x)}

#session options
options(stringsAsFactors = FALSE)

#set up the ggplot default params
theme_set(theme_bw(12) + 
          theme(panel.grid.major = element_blank(), 
                panel.grid.minor = element_blank(),
                plot.title = element_text(size=15, face="bold", margin = margin(10,0,10,0)),
                axis.text.x = element_text(angle=45, hjust = 1)))

#set up filename prefixes and output folders with the data
dir.create("Output")
dir.create(file.path("Output", "paper"))
dir.create(file.path("Output", "paper", format(Sys.Date(), "%y%m%d")))
dir.create(file.path("Output", "paper", format(Sys.Date(), "%y%m%d"), "R_objects"))
dir.create(file.path("Output", "paper", format(Sys.Date(), "%y%m%d"), "QC_plots"))
dir.create(file.path("Output", "paper", format(Sys.Date(), "%y%m%d"), "plots"))
dir.create(file.path("Output", "paper", format(Sys.Date(), "%y%m%d"), "pathways"))

fname_prefix_R = file.path("Output","paper", format(Sys.Date(), "%y%m%d"), "R_objects", format(Sys.Date(), "%y%m%d"))
fname_prefix_QC = file.path("Output","paper", format(Sys.Date(), "%y%m%d"), "QC_plots", format(Sys.Date(), "%y%m%d"))
fname_prefix_plots = file.path("Output","paper", format(Sys.Date(), "%y%m%d"), "plots", format(Sys.Date(), "%y%m%d"))
fname_prefix_pathways = file.path("Output","paper", format(Sys.Date(), "%y%m%d"), "pathways", format(Sys.Date(), "%y%m%d"))

#colors
col_B8 = "#D81B60"
col_M8 = "#1E88E5"
col_E8 = "#FFC107"
colors = c(col_B8, col_E8, col_M8)
```

#imported files: counts_ids, gene_info_filtered, meta

#mitochondrial genes
```{r}
#mt genes
mt_genes = c()
for(i in gene_info_filtered$gene_name) {x =  grepl("MT-", i); if(x == TRUE) {mt_genes = append(mt_genes, i)}}
mt_counts = counts_ids
mt_counts$gene_ID = rownames(mt_counts)
mt_counts = mt_counts %>% left_join(gene_info_filtered, by = "gene_ID")
mt_counts = mt_counts %>% dplyr :: filter(gene_name %in% mt_genes)
mt_counts = mt_counts %>% dplyr :: select(- c(gene_name, gene_type, gene_ID))
mt_counts = as.matrix(mt_counts)

mt_counts_sum = as.data.frame(colSums(mt_counts))
colnames(mt_counts_sum) = "mt_genes_sum"
mt_counts_sum$sample_ID = rownames(mt_counts_sum)
meta = meta  %>% left_join(mt_counts_sum, by = "sample_ID")

counts_names_matrix = as.matrix(counts_ids)
all_counts_sum = as.data.frame(colSums(counts_names_matrix))
colnames(all_counts_sum) = "all_genes_sum"
all_counts_sum$sample_ID = rownames(all_counts_sum)
meta = meta  %>% left_join(all_counts_sum, by = "sample_ID")

meta = meta %>% dplyr :: mutate(mt_perc = mt_genes_sum/all_genes_sum*100)
ggplot(meta, aes(x = sample_ID, y = mt_perc, fill = medium)) + geom_bar(stat = "identity") + scale_fill_manual(values = colors)
ggsave(filename = paste0(fname_prefix_QC, "_mt_perc.pdf"),
       width = 5, height = 4)
```

#QC
```{r}
#create DESeqDataSet 
dds = make_dds(counts = counts_ids, metadata = meta, ah_record = "AH89426")

#Total sample counts
plot_total_counts(dds)
ggsave(filename = paste0(fname_prefix_QC, "_counts_distribution.pdf"),
       width = 5, height = 4)

#Library complexity
plot_library_complexity(dds)
ggsave(filename = paste0(fname_prefix_QC, "_library_complexity.pdf"),
       width = 5, height = 4)

#Gene detection
plot_gene_detection(dds)
ggsave(filename = paste0(fname_prefix_QC, "_gene_detection.pdf"),
       width = 5, height = 4)

#Gene biotypes
plot_biotypes(dds)
ggsave(filename = paste0(fname_prefix_QC, "_biotypes.pdf"),
       width = 5, height = 4)

#Variance stabilization
vsd = vst(dds)
mean_sd_plot(vsd)
ggsave(filename = paste0(fname_prefix_QC, "_variance.pdf"),
       width = 5, height = 4)

#Chromosomal expression
chromosomes = c(1:22, "X", "Y", "MT")
for(i in chromosomes) {as.ggplot(plot_chromosome(vsd, as.character(i))); ggsave(filename = paste0(fname_prefix_QC, "_ch", as.character(i), ".pdf"),
       width = 5, height = 4)}

#Replicate variability
plot_ma = plot_sample_MAs(vsd, group = "medium")
for(i in 1:length(plot_ma)) {as.ggplot(plot_ma[[i]]); ggsave(filename = paste0(fname_prefix_QC, plot_ma[[i]][["labels"]][["title"]], ".pdf"),
       width = 5, height = 4)}

#Batch effect
vsd2 = vsd
assay(vsd2) = limma::removeBatchEffect(assay(vsd2), vsd2$rep)

#Clustering
as.ggplot(plot_sample_clustering(vsd, n_feats = 1000, anno_vars = c("medium", "rep"), distance = "pearson")) + ggtitle(" Pearson correlation") + theme(plot.title = element_text(face = "bold"))
ggsave(filename = paste0(fname_prefix_QC, "_clustering1.pdf"),
       width = 6, height = 4)
as.ggplot(plot_sample_clustering(vsd2, n_feats = 1000, anno_vars = c("medium", "rep"), distance = "pearson")) + ggtitle(" Pearson correlation") + theme(plot.title = element_text(face = "bold"))
ggsave(filename = paste0(fname_prefix_QC, "_clustering2.pdf"),
       width = 6, height = 4)
```

#PCA
```{r}
#pca before batch correction
plot_pca_scatters(vsd, n_PCs = 9,  color_by = "medium", shape_by = "rep")
ggsave(filename = paste0(fname_prefix_QC, "_PCA_1to9.pdf"),
       width = 18, height = 12)
pca_res1 = plot_pca(vsd, show_plot = FALSE, n_feats = 1000)
for(i in 1:9) {plot_loadings(pca_res1, PC = i, annotate_top_n = 5); ggsave(filename = paste0(fname_prefix_QC, "_pca_top5_", as.character(i), ".pdf"), width = 5, height = 4)}
for(i in 1:9) {plot_loadings(pca_res1, PC = i, color_by = "gene_biotype", show_plot = F)$plot +
    theme(legend.position = "bottom"); ggsave(filename = paste0(fname_prefix_QC, "_pca_biotypes_", as.character(i), ".pdf"), width = 5, height = 4)}
for(i in 1:9) {plot_loadings(pca_res1, PC = i, color_by = "gc_content"); ggsave(filename = paste0(fname_prefix_QC, "_pca_gc_", as.character(i), ".pdf"), width = 5, height = 4)}
pca_var1 = as.data.frame(pca_res1[["var_exp"]])
colnames(pca_var1) = "pca_var"
pca_var1$pca = c(1:9)
ggplot(pca_var1, aes(x = pca, y = pca_var)) + geom_point()
ggsave(filename = paste0(fname_prefix_QC, "_scree_plot1.pdf"),
       width = 5, height = 4)
pca1 = pca_res1[["data"]]
pca1$replicate = pca1$rep
ggplot(pca1, aes(x = PC1, y = PC2, shape = replicate)) + geom_point(aes(colour = medium), size = 4) + scale_color_manual(values = colors) +
  xlab(paste0("PC1 ", "60,43", " % variability")) + ylab(paste0("PC2 ", "17,33", " % variability")) + geom_point(colour = "grey90", size = 1.5) + ggtitle("PCA for the 1000 most variable genes")
ggsave(filename = paste0(fname_prefix_QC, "_PCA_1_2.pdf"),
       width = 7, height = 4)

#pca after batch correction
plot_pca_scatters(vsd2, n_PCs = 9,  color_by = "medium", shape_by = "rep")
ggsave(filename = paste0(fname_prefix_QC, "_PCA_1to9_corr.pdf"),
       width = 18, height = 12)
pca_res2 = plot_pca(vsd2, show_plot = FALSE, n_feats = 1000)
for(i in 1:9) {plot_loadings(pca_res2, PC = i, annotate_top_n = 5); ggsave(filename = paste0(fname_prefix_QC, "_pca_top5_", as.character(i), "corr.pdf"), width = 5, height = 4)}
for(i in 1:9) {plot_loadings(pca_res2, PC = i, color_by = "gene_biotype", show_plot = F)$plot +
    theme(legend.position = "bottom"); ggsave(filename = paste0(fname_prefix_QC, "_pca_biotypes_", as.character(i), "corr.pdf"), width = 5, height = 4)}
for(i in 1:9) {plot_loadings(pca_res2, PC = i, color_by = "gc_content"); ggsave(filename = paste0(fname_prefix_QC, "_pca_gc_", as.character(i), "corr.pdf"), width = 5, height = 4)}
pca_var2 = as.data.frame(pca_res2[["var_exp"]])
colnames(pca_var2) = "pca_var"
pca_var2$pca = c(1:9)
ggplot(pca_var2, aes(x = pca, y = pca_var)) + geom_point()
ggsave(filename = paste0(fname_prefix_QC, "_scree_plot2.pdf"),
       width = 5, height = 4)
pca2 = pca_res2[["data"]]
pca2$replicate = pca1$rep
ggplot(pca2, aes(x = PC1, y = PC2, shape = replicate)) + geom_point(aes(colour = medium), size = 4) + scale_color_manual(values = colors) +
  xlab(paste0("PC1 ", "71,71", " % variability")) + ylab(paste0("PC2 ", "14,38", " % variability")) + geom_point(colour = "grey90", size = 1.5) + ggtitle("PCA for the 1000 most variable genes")
ggsave(filename = paste0(fname_prefix_QC, "_PCA_1_2_corr.pdf"),
       width = 7, height = 4)
```

#limma
```{r}
#differential expression analysis
media = meta$medium
rep = meta$rep
design = model.matrix(~0+media)
contrasts = makeContrasts(E8vsB8 = mediaE8 - mediaB8, B8vsM8 = mediaB8 - mediaM8, M8vsE8 = mediaM8 - mediaE8, levels = colnames(design))
v = voom(counts_ids, design, plot = TRUE)
v[["E"]] = limma::removeBatchEffect(v[["E"]], rep)
l = lmFit(v, design)
vfit = contrasts.fit(l, contrasts = contrasts)
efit = eBayes(vfit)
plotSA(efit)

diff_E8vsB8 = topTreat(efit, coef=1, n = Inf)
diff_B8vsM8 = topTreat(efit, coef=2, n = Inf)
diff_M8vsE8 = topTreat(efit, coef=3, n = Inf)

#E8 vs B8
diff_E8vsB8$gene_ID = rownames(diff_E8vsB8)
diff_E8vsB8 = diff_E8vsB8 %>% left_join(gene_info_filtered, by = "gene_ID")

#M8 vs E8
diff_M8vsE8$gene_ID = rownames(diff_M8vsE8)
diff_M8vsE8 = diff_M8vsE8 %>% left_join(gene_info_filtered, by = "gene_ID")

#B8 vs others
B8 = c()
for(i in media){if(i == "B8"){B8 = append(B8, "B8")} else{B8 = append(B8, "others")}}
designB8 = model.matrix(~0+B8)
contrastsB8 = makeContrasts(B8 = B8B8 - B8others, levels = colnames(designB8))
v_B8 = voom(counts_ids, designB8, plot = TRUE)
v_B8[["E"]] = limma::removeBatchEffect(v_B8[["E"]], rep)
l_B8 = lmFit(v_B8, designB8)
vfitB8 = contrasts.fit(l_B8, contrasts = contrastsB8)
efitB8 = eBayes(vfitB8)
plotSA(efitB8)
diff_B8 = topTreat(efitB8, coef = 1, n = Inf)
diff_B8$gene_ID = rownames(diff_B8)
diff_B8 = diff_B8 %>% left_join(gene_info_filtered, by = "gene_ID")

#E8 vs others
E8 = c()
for(i in media){if(i == "E8"){E8 = append(E8, "E8")} else{E8 = append(E8, "others")}}
designE8 = model.matrix(~0+E8)
contrastsE8 = makeContrasts(E8 = E8E8 - E8others, levels = colnames(designE8))
v_E8 = voom(counts_ids, designE8, plot = TRUE)
v_E8[["E"]] = limma::removeBatchEffect(v_E8[["E"]], rep)
l_E8 = lmFit(v_E8, designE8)
vfitE8 = contrasts.fit(l_E8, contrasts = contrastsE8)
efitE8 = eBayes(vfitE8)
plotSA(efitE8)
diff_E8 = topTreat(efitE8, coef = 1, n = Inf)
diff_E8$gene_ID = rownames(diff_E8)
diff_E8 = diff_E8 %>% left_join(gene_info_filtered, by = "gene_ID")

#M8 vs others
M8 = c()
for(i in media){if(i == "M8"){M8 = append(M8, "M8")} else{M8 = append(M8, "others")}}
designM8 = model.matrix(~0+M8)
contrastsM8 = makeContrasts(M8 = M8M8 - M8others, levels = colnames(designM8))
v_M8 = voom(counts_ids, designM8, plot = TRUE)
v_M8[["E"]] = limma::removeBatchEffect(v_M8[["E"]], rep)
l_M8 = lmFit(v_M8, designM8)
vfitM8 = contrasts.fit(l_M8, contrasts = contrastsM8)
efitM8 = eBayes(vfitM8)
plotSA(efitM8)
diff_M8 = topTreat(efitM8, coef = 1, n = Inf)
diff_M8$gene_ID = rownames(diff_M8)
diff_M8 = diff_M8 %>% left_join(gene_info_filtered, by = "gene_ID")
```

#normalization
```{r}
#counts
#upload gene_lenght file
counts_tpm = as.data.frame(ADImpute::NormalizeTPM(as.matrix(counts_ids), sce = NULL, log = FALSE, tr_length = gene_length, scale = 1))
counts_tpm = as.data.frame(limma::removeBatchEffect(counts_tpm, meta$rep))
```

#WGCNA
```{r}
#wgcna
allowWGCNAThreads()
powers = c(1:20)
counts_w = t(counts_tpm)
sft = pickSoftThreshold(counts_w, powerVector = powers, verbose = 5)
plot(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     xlab = "Soft Threshold (power)",
     ylab = "Scale Free Topology Model Fit, signed R^2",
     main = paste("Scale independence"))
picked_power = 9
netwk = blockwiseModules(counts_w,                
                          power = picked_power, 
                          networkType = "signed",
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          minModuleSize = 30,
                          maxBlockSize = 4000,
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,
                          saveTOMs = T,
                          saveTOMFileBase = "ER",
                          numericLabels = T,
                          verbose = 3)

mergedColors = labels2colors(netwk$colors)

module = data.frame(
  gene_ID = names(netwk$colors),
  colors = mergedColors)

MEs0 = moduleEigengenes(counts_w, mergedColors)$eigengenes
as.ggplot(pheatmap(MEs0, scale = "column", color = viridis::viridis(n = 25, option = "H")))
ggsave(filename = paste0(fname_prefix_plots, "_heatmap_modules.pdf"),
       width = 7, height = 4)

module = module %>% dplyr::left_join(gene_info_filtered, by = "gene_ID")
```

#significant genes in modules
```{r}
#selection of significative genes
e8b8 = c()
for(i in 1:length(rownames(diff_E8vsB8))){if(diff_E8vsB8$adj.P.Val[i] < 0.05) {e8b8 = append(e8b8, diff_E8vsB8$gene_name[i])}}
e8m8 = c()
for(i in 1:length(rownames(diff_M8vsE8))){if(diff_M8vsE8$adj.P.Val[i] < 0.05) {e8m8 = append(e8m8, diff_M8vsE8$gene_name[i])}}
diff = unique(c(e8b8, e8m8))
module_sign = module %>% dplyr::filter(gene_name %in% diff)
module_sign = module_sign %>% arrange(colors)
counts = counts_tpm
counts$gene_ID = rownames(counts)
counts = counts %>% dplyr::left_join(gene_info_filtered, by = "gene_ID")
counts = counts %>% dplyr::select(-(c(gene_ID, gene_type)))
counts_diff = counts %>% dplyr::filter(gene_name %in% diff) %>% dplyr::left_join(module_sign, by = "gene_name") %>% dplyr::arrange(colors)

#heatmap
rownames(counts_diff) = counts_diff$gene_name
counts_diff = counts_diff %>% dplyr::select(meta$sample_ID)
ann_col = meta
rownames(ann_col) = ann_col$sample_ID
ann_col = ann_col %>% dplyr::select(medium)
ann_row = module_sign
rownames(ann_row) = ann_row$gene_name
ann_row = ann_row %>% dplyr::select(colors)
colors = unique(module$colors)
names(colors) = colors
ann_colors = list(medium = c(B8 = col_B8, E8 = col_E8, M8 = col_M8), colors = colors)

as.ggplot(pheatmap(as.matrix(counts_diff), scale = "row", cluster_rows = FALSE, show_rownames = FALSE, annotation_col = ann_col, annotation_row = ann_row, annotation_legend = FALSE, annotation_names_col = TRUE, annotation_names_row = TRUE, annotation_colors = ann_colors, cutree_cols = 3, color = viridis::viridis(n = 25, option = "H"))) + ggtitle(" Differentially expressed genes") + theme(plot.title = element_text(face = "bold"))
ggsave(filename = paste0(fname_prefix_plots, "_heatmap_diff_mod2.pdf"),
       width = 7, height = 4)

#B8 -> brown
brown = module_sign %>% dplyr::filter(colors == "brown")
GO = ensembldb::select(hs, keys = brown$gene_name, columns = c("ENTREZID", "SYMBOL"), keytype = "SYMBOL")
go = goana(GO$ENTREZID, species = "Hs", convert = T)
BP_B8 = topGO(go, ontology = c("BP"), number = 50, truncate.term = 50)
ggplot(BP_B8, aes(x = reorder(Term, +DE), fill = - log10(BP_B8$P.DE), y = DE)) + geom_bar(stat = 'identity', width = 0.8) + coord_flip() +
  labs (title = "GO brown", colour = "-Log10Pval") + xlab("") + ylab("n of genes") 
ggsave(filename = paste0(fname_prefix_plots, paste0("_BP_top_brown_.pdf")),
  width = 10, height = 4)

#E8 -> turquoise
turquoise = module_sign %>% dplyr::filter(colors == "turquoise")
GO = ensembldb::select(hs, keys = turquoise$gene_name, columns = c("ENTREZID", "SYMBOL"), keytype = "SYMBOL")
go = goana(GO$ENTREZID, species = "Hs", convert = T)
BP_E8 = topGO(go, ontology = c("BP"), number = 50, truncate.term = 50)
ggplot(BP_E8, aes(x = reorder(Term, +DE), fill = - log10(BP_E8$P.DE), y = DE)) + geom_bar(stat = 'identity', width = 0.8) + coord_flip() +
  labs (title = "GO turquoise", colour = "-Log10Pval") + xlab("") + ylab("n of genes") 
ggsave(filename = paste0(fname_prefix_plots, paste0("_BP_top_turquoise_.pdf")),
  width = 10, height = 4)

#M8 -> blue
blue = module_sign %>% dplyr::filter(colors == "blue")
GO = ensembldb::select(hs, keys = blue$gene_name, columns = c("ENTREZID", "SYMBOL"), keytype = "SYMBOL")
go = goana(GO$ENTREZID, species = "Hs", convert = T)
BP_M8 = topGO(go, ontology = c("BP"), number = 50, truncate.term = 50)
ggplot(BP_M8, aes(x = reorder(Term, +DE), fill = - log10(BP_M8$P.DE), y = DE)) + geom_bar(stat = 'identity', width = 0.8) + coord_flip() +
  labs (title = "GO blue", colour = "-Log10Pval") + xlab("") + ylab("n of genes") 
ggsave(filename = paste0(fname_prefix_plots, paste0("_BP_top_blue_.pdf")),
  width = 10, height = 4)
```

#germ layers gsea
```{r}
organism = "org.Hs.eg.db"
library(organism, character.only = TRUE)
set.seed(1234) 

#B8
genesB8vsothers = diff_B8$logFC
names(genesB8vsothers) = diff_B8$gene_ID
genesB8vsothers = sort(genesB8vsothers, decreasing = TRUE)
gseB8 <- gseGO(geneList = genesB8vsothers, 
             ont ="BP", 
             keyType = "ENSEMBL",
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 1, 
             verbose = TRUE, 
             OrgDb = organism, 
             pAdjustMethod = "BH")
resB8 = gseB8@result
resB8$n = c(1:length(rownames(resB8)))
#neural
gseaplot(gseB8, by ="runningScore", title = gseB8$Description[421], geneSetID = 421, color.line = col_B8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_neuroB8.pdf"),
       width = 5, height = 3)
#endoderm
gseaplot(gseB8, by ="runningScore", title = gseB8$Description[29], geneSetID = 29, color.line = col_B8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_endoB8.pdf"),
       width = 5, height = 3)
#mesoderm
gseaplot(gseB8, by ="runningScore", title = gseB8$Description[171], geneSetID = 171, color.line = col_B8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_mesoB8.pdf"),
       width = 5, height = 3)

#E8
genesE8vsothers = diff_E8$logFC
names(genesE8vsothers) = diff_E8$gene_ID
genesE8vsothers = sort(genesE8vsothers, decreasing = TRUE)
gseE8 <- gseGO(geneList = genesE8vsothers, 
             ont ="BP", 
             keyType = "ENSEMBL",
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 1, 
             verbose = TRUE, 
             OrgDb = organism, 
             pAdjustMethod = "BH")
resE8 = gseE8@result
resE8$n = c(1:length(rownames(resE8)))
#neural
gseaplot(gseE8, by ="runningScore", title = gseE8$Description[796], geneSetID = 796, color.line = col_E8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_neuroE8.pdf"),
       width = 5, height = 3)
#endoderm
gseaplot(gseE8, by ="runningScore", title = gseE8$Description[40], geneSetID = 40, color.line = col_E8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_endoE8.pdf"),
       width = 5, height = 3)
#mesoderm
gseaplot(gseE8, by ="runningScore", title = gseE8$Description[72], geneSetID = 72, color.line = col_E8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_mesoE8.pdf"),
       width = 5, height = 3)

#M8
genesM8vsothers = diff_M8$logFC
names(genesM8vsothers) = diff_M8$gene_ID
genesM8vsothers = sort(genesM8vsothers, decreasing = TRUE)
gseM8 <- gseGO(geneList = genesM8vsothers, 
             ont ="BP", 
             keyType = "ENSEMBL",
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 1, 
             verbose = TRUE, 
             OrgDb = organism, 
             pAdjustMethod = "BH")
resM8 = gseM8@result
resM8$n = c(1:length(rownames(resM8)))
#neural
gseaplot(gseM8, by ="runningScore", title = gseM8$Description[4537], geneSetID = 4537, color.line = col_M8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_neuroM8.pdf"),
       width = 5, height = 3)
#endoderm
gseaplot(gseM8, by ="runningScore", title = gseM8$Description[1466], geneSetID = 1466, color.line = col_M8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_endoM8.pdf"),
       width = 5, height = 3)
#mesoderm
gseaplot(gseM8, by ="runningScore", title = gseM8$Description[5200], geneSetID = 5200, color.line = col_M8) + theme(plot.title = element_text(face = "bold")) + scale_y_continuous(limits = c(-0.7, 0.7))
ggsave(filename = paste0(fname_prefix_plots, "_mesoM8.pdf"),
       width = 5, height = 3)
```

#pluripotency pathways
```{r}
#Kegg
kegg_organism = "hsa"
set.seed(1234)

#E8 vs B8
E8vsB8 = diff_E8vsB8 %>% dplyr::filter(adj.P.Val < 0.05)
genesE8vsB8 = E8vsB8$logFC
names(genesE8vsB8) = E8vsB8$gene_ID

ids = bitr(names(genesE8vsB8), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=organism)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]
E8vsB82 = E8vsB8[E8vsB8$gene_ID %in% dedup_ids$ENSEMBL,]
E8vsB82$K_id = dedup_ids$ENTREZID
keggE8vsB8 = E8vsB82$logFC * (-1)
names(keggE8vsB8) = E8vsB82$K_id
keggE8vsB8 = na.omit(keggE8vsB8)
keggE8vsB8 = sort(keggE8vsB8, decreasing = TRUE)
kkE8vsB8 <- gseKEGG(geneList = keggE8vsB8,
               organism = kegg_organism,
               minGSSize = 3,
               maxGSSize = 800,
               pvalueCutoff = 1,
               pAdjustMethod = "BH",
               keyType = "kegg")
reskE8vsB8 = kkE8vsB8@result
reskE8vsB8$n = c(1:length(rownames(reskE8vsB8)))

#M8 vs E8
M8vsE8 = diff_M8vsE8 %>% dplyr::filter(adj.P.Val < 0.05)
genesM8vsE8 = M8vsE8$logFC
names(genesM8vsE8) = M8vsE8$gene_ID

ids = bitr(names(genesM8vsE8), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=organism)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]
M8vsE82 = M8vsE8[M8vsE8$gene_ID %in% dedup_ids$ENSEMBL,]
M8vsE82$K_id = dedup_ids$ENTREZID
keggM8vsE8 = M8vsE82$logFC
names(keggM8vsE8) = M8vsE82$K_id
keggM8vsE8 = na.omit(keggM8vsE8)
keggM8vsE8 = sort(keggM8vsE8, decreasing = TRUE)
kkM8vsE8 <- gseKEGG(geneList = keggM8vsE8,
               organism = kegg_organism,
               minGSSize = 3,
               maxGSSize = 800,
               pvalueCutoff = 1,
               pAdjustMethod = "BH",
               keyType = "kegg")
reskM8vsE8 = kkM8vsE8@result
reskM8vsE8$n = c(1:length(rownames(reskM8vsE8)))


#pluripotency kegg pathway
pluripotencyID = "hsa04550"
pathB8vsE8 = pathview(gene.data = keggE8vsB8, pathway.id = pluripotencyID, species = kegg_organism, low = list(gene = col_E8, cpd = col_E8), high = list(gene = col_B8, cpd = col_B8))
pathM8vsE8 = pathview(gene.data = keggM8vsE8, pathway.id = pluripotencyID, species = kegg_organism, low = list(gene = col_E8, cpd = col_E8), high = list(gene = col_M8, cpd = col_M8))

#pluripotency markers
markers = c("SOX2", "POU5F1", "NANOG")
melt_plur = melt_norm %>% dplyr::filter(gene_name %in% markers)
order = c()
for(i in melt_plur$medium){if(i == "E8"){order = append(order, "a")} else if(i == "M8"){order = append(order, "b")} else if(i == "B8"){order = append(order, "c")}}
melt_plur$order = order
ggplot(melt_plur) + geom_boxplot(aes(x = order, y = value, fill = medium)) + facet_wrap(~ gene_name) + scale_fill_manual(values = c(col_B8, col_E8, col_M8))
ggsave(filename = paste0(fname_prefix_plots, "_boxplot_pluripotency.pdf"),
       width = 12, height = 4)
```
